raiden-ts/src/services/epics.ts

```diff
@@ -442,15 +445,20 @@ export function pfsServiceRegistryMonitorEpic(
  * @returns Observable of udcDeposited actions
  */
 export function monitorUdcBalanceEpic(
-  {}: Observable<RaidenAction>,
+  action$: Observable<RaidenAction>,
   {}: Observable<RaidenState>,
   { address, latest$, userDepositContract }: RaidenEpicDeps,
 ): Observable<udcDeposit.success> {
-  return latest$.pipe(
-    pluckDistinct('state', 'blockNumber'),
+  return action$.pipe(
+    filter(newBlock.is),
+    startWith(null),
     // it's seems ugly to call on each block, but UserDepositContract doesn't expose deposits as
     // events, and ethers actually do that to monitor token balances, so it's equivalent
-    exhaustMap(() => userDepositContract.functions.effectiveBalance(address) as Promise<UInt<32>>),
+    exhaustMap(() =>
+      defer(
+        async () => userDepositContract.functions.effectiveBalance(address) as Promise<UInt<32>>,
+      ).pipe(catchError(constant(EMPTY))),
```

Comment:
Does that mean we simply ignore errors? :scream:

Reply:
Yes, we do, since these errors should be just network-related (not something triggered by the user, or so), and it's a retriable error by nature, therefore we just omit any error here, and on next block it'll get requested again: notice that, if this was caused by connectivity loss (as I've witnessed), no new blocks notification will go through, ethers loop omit them and will re-emit when connectivity is restored, therefore these errors should happen only in case of a race (`newBlock` was emitted, but at the time `effectiveBalance` was requested, it got disconnected).
I'll write a comment to describe it better. Thanks

Reply:
Could you eventually do a small new PR and add some logging to it? I think it is fine to handle it like this. But for the (uncertain?) case that there could be a problem at a later point, it would be cool to at least see this in the logs while investigating on errors.

Reply:
This is a retried error, which is not user-triggered. I don't think logs are useful here, specially because it triggers only on race-conditions (current blockNumber call succeeded, function call failed). A more useful logging would be detecting we are missing blocks for some time, possibly even halting receiving transfers => #1256